core/src/com/unciv/logic/automation/Automation.kt

```diff
@@ -116,16 +116,31 @@ object Automation {
                 newGrowthFood += growthFood / 4
             }
             newGrowthFood = newGrowthFood.coerceAtLeast(0f) // floor to 0 for safety
-
-            // When Happy, 2 production is better than 1 growth,
-            // but setting such by default worsens AI civ citizen assignment,
-            // probably due to badly configured personalities not properly weighing food vs non-food yields
-
-            // Zero out Growth if close to Unhappiness limit as well
-            val baseFocusWeight = if (city.civ.getHappiness() < -8) 0 else {
-                if (cityAIFocus in CityFocus.zeroFoodFocuses) 1 else 2
+            
+            
+            var foodModWeight = 1f
```

Comment:
This is already 20 lines of dense specific logic to define a single value, please make this a getFoodModWeight() function